File: NOVEMBER-2025/28-11-2025/Subset XOR.js
class Solution {
    subsetXOR(n) {
        let totalXOR = 0;
        for (let i = 1; i <= n; i++) totalXOR ^= i;
        if (totalXOR === n) {
            return Array.from({length: n}, (_, i) => i + 1);
        } else {
            const x = totalXOR ^ n;
            let ans = [];
            for (let i = 1; i <= n; i++) {
                if (i !== x) ans.push(i);
            }
            return ans;
        }
    }
}

File: NOVEMBER-2025/28-11-2025/Subset XOR.cs
class Solution {
    public List<int> SubsetXOR(int n) {
        List<int> ans = new List<int>();
        int totalXOR = 0;
        for (int i = 1; i <= n; i++) totalXOR ^= i;
        if (totalXOR == n) {
            for (int i = 1; i <= n; i++) ans.Add(i);
        } else {
            int x = totalXOR ^ n;
            for (int i = 1; i <= n; i++) {
                if (i != x) ans.Add(i);
            }
        }
        return ans;
    }
}